Tools/cases_generator/generate_cases.py

```diff
@@ -61,12 +62,18 @@ def always_exits(block: parser.Block) -> bool:
 
 
 def write_cases(f: io.TextIOBase, instrs: list[InstDef]):
+    predictions = set()
+    for inst in instrs:
+        for _, target in re.findall(r"(PREDICT|GO_TO_INSTRUCTION)\((\w+)\)", inst.block.text):
```

Comment:
You can use a non-capturing group here:

```suggestion
        for target in re.findall(r"(?:PREDICT|GO_TO_INSTRUCTION)\((\w+)\)", inst.block.text):
```